exaudfclient/docs/script_options_requirments.md

```diff
@@ -0,0 +1,330 @@
+
+Needs: dsn
+
+Tags: V1
+
+Covers:
+- `feat~java-specific-script-options~1`
+
+### Java %jar Option Handling V2
+`req~java-jar-option-handling-v2~1`
+
+The Java parser handler must find multiple %jar options. The values are to be interpreted as the Java CLASSPATH: `<file1>:<file2>:...:<filen>`.
+The Java parser handler shall split the entries by the colon character. The Java parser handler must keep duplicates. The order of the entries must not change.
+
+Needs: dsn
+
+Tags: V2
+
+Covers:
+- `feat~java-specific-script-options~1`
+
+### Java %jar Option Trailing White Space Handling
+`req~java-jar-option-trailing-white-space-handling~1`
+
+The Java parser handler must remove trailing white spaces for `%jar` option values if they are part of the escape sequence '\ '. Escape sequences at the end of a found `%jar` option of the form `\ ` must be replaced with ' '.
+This approach provides backwards compatibility for most existing UDF's from customers.
+
+Needs: dsn
+
+Tags: V2
+
+Covers:
+- `feat~java-specific-script-options~1`
+
+Depends: 
+- `req~white-spaces-script-options-parsing-v2~1`
+
+### Java %jvmoption Handling
+`req~java-jvmoption-handling~1`
+
+The Java parser handler must find multiple %jvmoption options, allowing duplicates and maintaining order.
+
+Needs: dsn
+
+Covers:
+- `feat~java-specific-script-options~1`
+
+### Java %import Option Replace Referenced Sripts
+`req~java-import-option-replace-referenced-scripts~1`
+
+For each found %import option, the Java parser handler must request and replace the referenced scripts recursively. This means,
+if the referenced scripts contain also `%import` options, the implementation must replace those, too.
+The referenced script name should be handled according to the [Exasol SQL identifier specification](https://docs.exasol.com/db/latest/sql_references/basiclanguageelements.htm#SQLidentifier).
+
+Needs: dsn
+
+Covers:
+- `feat~java-specific-script-options~1`
+
+### Java %import Option Handling
+`req~java-import-option-handling~1`
+
+For each found %import option, the Java parser handler must handle nested Script Options appropriately:
+1. `%scriptclass` option must be ignored, but removed from the script code.
+2. All other options must be handled as if they were part of the source script.
+
+Needs: dsn
+
+Covers:
+- `feat~java-specific-script-options~1`
+
+### Existing Parser Library Dependencies
+`req~existing-parser-library-dependencies~1`
+
+The new parser must be implemented using an existing, open-source parser that supports definition of Lexer and Parser Rules in C++ code without additional runtime dependencies.
+The implementation needs to be Open Source because the projects where the parser will be used are mainly Open Source, too. 
+It is important to avoid additional runtime dependencies, as this would complicate the setup and maintenance of the runtime environment of the UDF client (aka Script Languages Container).
```

Comment:
We are also limited to certain licenses that can also used with closed source. So, a permissive license without copy left.

Reply:
Done